### PR TITLE
Make sure all relevant fields are queried on school info deduping

### PR DESCRIPTION
**Pull request by drewsamnick**
This fixes issue that when we dedupe we were throwing out most of the blank parameters from the query. The effect is that we treat those fields as wildcards rather than only matching null values. For example, if you tried to create a school info with only the country set to US then it would match the first row found with country US *even if other fields are also set*. So you match a specific school that you did not enter. In the most extreme case, trying to create a school info with none of the fields set then it will successfully match the first row with validation type "none" regardless of what values are set in the other fields.

To fix this, I added code to ensure that all of the relevant fields are present when we make the query. I updated the existing tests and add one specifically to test that we do not find a match in these cases. That test fails without my changes.

### CODE REVIEW

File: dashboard/test/lib/school_info_deduplicator_test.rb [hunk 1]
```diff
@@ -139,4 +166,15 @@ class MockSchoolInfoDeduplicator
     assert deduped, "Expected to dedupe on school id"
     assert_equal school_info.id, mock.school_info.id
   end
+
+  test 'missing or blank fields do not match none-null values' do
+    mock = MockSchoolInfoDeduplicator.new
```
Comment (aoby) [inline dashboard/test/lib/school_info_deduplicator_test.rb]: nit: I see this is following an existing pattern, so no need to change, but I would prefer naming this variable something more contextual and descriptive like `deduplicator` or `mockDeduplicator`.  The variable name `mock` is unhelpful, like calling a variable `instance` or `data`.

**Comment by drewsamnick** [comment]
@aoby Yes, the test I added covers the previously broken scenario.

File: dashboard/lib/school_info_deduplicator.rb [hunk 2]
```diff
@@ -70,7 +58,15 @@ def process_school_info_attributes(school_info_attr)
       :school_name,
       :full_address,
       :validation_type
-    )
+    ]
+
+    attr.slice!(*relevant_attributes)
```
Comment (breville) [inline dashboard/lib/school_info_deduplicator.rb]: I wonder if we can save some folks a quick google and add a brief comment that this only leaves the provided keys intact?

**Comment by breville** [comment]
Worth noting here that this must mean we have incorrectly de-duped data in production.

**Comment by bencodeorg** [comment]
We definitely do (e.g., we have a lot of afterschool teachers linked to a school named "Eric's awesome school" or something). 

On the plus side, the only teachers affected are those who did not give us any information that allows us to link them to a schematized NCES school ID (which is really the only type of data that we use currently). If we want to use that free-form text in any way then we'd need to do some sort of cleanup (I did something similar when we noticed a similar bug a couple months ago, so could help out with that).

**Comment by ewjordan** [comment]
Ah yeah, this was something that we realized a while back, I remember discussing it with *someone* (at least Ben, but I thought an engineer was in the loop and working on a fix, I might be wrong), but can't find the records now. Good fix!

**Comment by bencodeorg** [comment]
Joanne fixed for cases where teachers provided and/or could be linked a specific school with a NCES ID. My understanding is this fixes for other cases, but @drewsamnick weigh in if I'm mistaken.

File: dashboard/lib/school_info_deduplicator.rb (hunk 2 above)
Comment (drewsamnick) [reply dashboard/lib/school_info_deduplicator.rb]: Added a comment.

File: dashboard/test/lib/school_info_deduplicator_test.rb (hunk 1 above)
Comment (drewsamnick) [reply dashboard/test/lib/school_info_deduplicator_test.rb]: I went ahead and renamed this.

**Comment by drewsamnick** [comment]
I found another case that needed to be handled. The validation_type will default to 'full' if not provided. I fixed the code to dedupe on that value rather than null when it isn't provided. I also added a test cases for this.

**Pull request merged**